### DISCUSSION TITLE
What is the Environment Filter looking for in the Actor Stat Block?

### DISCUSSION THREAD

**Discussion by FoxMaskCain** [etriebe/dnd-randomizer#101] [Q&A] [answered]
I'm trying to add custom monsters to the encounter generator, and I want to make use of the environment filter, but I can't figure out what the filter is looking for in the Actor stat block. I can't find an "Environment" field in the Actor stat block, so is there a way to assign Environments on the module side to monsters within any compendium?

**Comment by etriebe** [comment]
Are you using pf2e or dnd5e?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: FoxMaskCain ***@***.***>
Sent: Saturday, March 30, 2024 2:17:24 PM
To: etriebe/dnd-randomizer ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [etriebe/dnd-randomizer] What is the Environment Filter looking for in the Actor Stat Block? (Discussion #101)


I'm trying to add custom monsters to the encounter generator, and I want to make use of the environment filter, but I can't figure out what the filter is looking for in the Actor stat block. I can't find an "Environment" field in the Actor stat block, so is there a way to assign Environments on the module side to monsters within any compendium?

**Reply by FoxMaskCain** [reply]
dnd 5e

**Comment by etriebe** [comment]
It’s normally right below some fields on the upper left of the dialog, afaik.

Just reference some SRD creatures and see where that shows up on the dialog.

Unless they changed something with DnD5e 3.0 system and I need some updates. 😅 that’s entirely possible. I haven’t checked on this in a bit.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by FoxMaskCain** [reply]
I did a side-by-side of my custom NPC actor and a 5e compendium NPC actor, and didn't find any environment field. Looked through every tab and config menu. Didn't see anything in the upper left either.

**Comment by etriebe** [comment]
Ok. Let me dig in tonight. My guess is 3.0 likely changed something.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by etriebe** [comment, marked as answer]
Ah, just realized this variable is set in the actor payload but only visible if you use a different module sheet like tidy5e sheet.
